app/controllers/api/v1/notifications/policies_controller.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class Api::V1::Notifications::PoliciesController < Api::BaseController
+  before_action -> { doorkeeper_authorize! :read, :'read:notifications' }, only: :show
+  before_action -> { doorkeeper_authorize! :write, :'write:notifications' }, only: :update
+
+  before_action :require_user!
+  before_action :set_policy
+
+  def show
+    render json: @policy, serializer: REST::NotificationPolicySerializer
+  end
+
+  def update
+    @policy.update!(resource_params)
+    render json: @policy, serializer: REST::NotificationPolicySerializer
+  end
+
+  private
+
+  def set_policy
+    with_read_replica do
+      @policy = NotificationPolicy.find_or_initialize_by(account: current_account)
+      @policy.summarize!
+    end
+  end
```

Comment:
I think reading from replica then writing to the primary can result in old data being saved, and would cause calls in quick succession to be unreliable. This could be an issue if clients try to push updates as the user clicks checkboxes, rather than batching them.